libs/Wallet/include/Status/Wallet/NewWalletAccountController.h

```diff
@@ -0,0 +1,94 @@
+#pragma once
+
+#include "Status/Wallet/WalletAccount.h"
+#include "Status/Wallet/DerivedWalletAddress.h"
+
+#include <Helpers/QObjectVectorModel.h>
+
+#include <QQmlListProperty>
+#include <QtQmlIntegration>
+
+class QQmlEngine;
+class QJSEngine;
+
+namespace Status::Wallet {
+
+/// \note the folowing values are kept in sync \c selectedDerivedAddress, \c derivedAddressIndex and \c derivationPath
+///       and \c customDerivationPath; \see connascence.io/value
+class NewWalletAccountController: public QObject
+{
+    Q_OBJECT
+    QML_UNCREATABLE("C++ only")
+
+    Q_PROPERTY(QAbstractListModel* mainAccountsModel READ mainAccountsModel CONSTANT)
+
+    Q_PROPERTY(QAbstractItemModel* currentDerivedAddressModel READ currentDerivedAddressModel CONSTANT)
+    Q_PROPERTY(DerivedWalletAddress* selectedDerivedAddress READ selectedDerivedAddress WRITE setSelectedDerivedAddress NOTIFY selectedDerivedAddressChanged)
+    Q_PROPERTY(int derivedAddressIndex MEMBER m_derivedAddressIndex NOTIFY selectedDerivedAddressChanged)
+
+    Q_PROPERTY(QString derivationPath READ derivationPath WRITE setDerivationPath NOTIFY derivationPathChanged)
+    Q_PROPERTY(bool customDerivationPath MEMBER m_customDerivationPath NOTIFY customDerivationPathChanged)
+
+public:
+    using AccountsModel = Helpers::QObjectVectorModel<WalletAccount>;
+
+    /// \note On account creation \c accounts are updated with the newly created wallet account
+    NewWalletAccountController(std::shared_ptr<AccountsModel> accounts);
+
+    /// Called by QML engine to register the instance. QML takes ownership of the instance
+    static NewWalletAccountController *create(QQmlEngine *qmlEngine, QJSEngine *jsEngine);
+
+
+    QAbstractListModel *mainAccountsModel();
+    QAbstractItemModel *currentDerivedAddressModel();
```

Comment:
`const` both

Reply:
`m_mainAccounts` and `m_derivedAddress` are non-const lvalue members.